### ISSUE TITLE
Cross validation results inconsistency

### ISSUE THREAD

**Issue by ericZYZ** [RasaHQ/rasa#7454]
**Rasa version**: 2.0.2

**Rasa SDK version** (if used & relevant): 2.0.0

**Rasa X version** (if used & relevant): 0.33.2

**Python version**: 3.6.10

**Operating system** (windows, osx, ...): Linux

**Issue**:
Rasa cross validation test results inconsistency.

**Error (including full traceback)**:
I can't seem to fully understand the test reports generated by rasa cross-validation test. For example, when running
 `rasa test nlu -f 3 --cross-validation` I got the following results in the terminal:
```
2020-12-04 11:56:16 INFO     rasa.test  - CV evaluation (n=3)
2020-12-04 11:56:16 INFO     rasa.test  - Intent evaluation results
2020-12-04 11:56:16 INFO     rasa.nlu.test  - train Accuracy: 0.838 (0.014)
2020-12-04 11:56:16 INFO     rasa.nlu.test  - train F1-score: 0.850 (0.013)
2020-12-04 11:56:16 INFO     rasa.nlu.test  - train Precision: 0.875 (0.013)
2020-12-04 11:56:16 INFO     rasa.nlu.test  - test Accuracy: 0.465 (0.007)
2020-12-04 11:56:16 INFO     rasa.nlu.test  - test F1-score: 0.496 (0.014)
2020-12-04 11:56:16 INFO     rasa.nlu.test  - test Precision: 0.581 (0.013)
```
However when I look at the intent_report.json, I have a different set of results which don't match what I see in the terminal:
```
 "micro avg": {
    "precision": 0.8641975308641975,
    "recall": 0.3431372549019608,
    "f1-score": 0.4912280701754386,
    "support": 204
  },
  "macro avg": {
    "precision": 0.5474404761904761,
    "recall": 0.26728896103896105,
    "f1-score": 0.34139778479174765,
    "support": 204
  },
  "weighted avg": {
    "precision": 0.6639705882352941,
    "recall": 0.3431372549019608,
    "f1-score": 0.4326650976555365,
    "support": 204
  }
```
Another issue is that the number of "support" is different for each different number of folds. What does this "support" represent here?  

**Command or request that led to error**:
```
rasa test nlu -f 3 --cross-validation
```

**Content of configuration file (config.yml)** (if relevant):
```yml

```

**Content of domain file (domain.yml)** (if relevant):
```yml

```

**Comment by sara-tagger** [comment]
Thanks for raising this issue, @erohmensing will get back to you about it soon✨

###### Please also check out the [docs](https://rasa.com/docs/) and the [forum](https://forum.rasa.com/) in case your issue was raised there too 🤗

**Comment by koernerfelicia** [comment]
Hello @ericZYZ!
I don't **think** this is an error, but let me try to explain what's going on!

The different types of `avg` you see in the `intent_report.json` come from `sklearn.metrics`. You can read more about them in the documentation linked in the [method](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.classification_report.html)  that produces them [here](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.precision_recall_fscore_support.html). The values in `intent_report.json` are calculated from the `test` results, which are concatenated for each fold. 

The cumulative results we print in the terminal are a result of calculating `acc`, `f1`, `precision` for `train/test` for each fold and taking the mean (`np.mean`). (The term in parentheses is the standard deviation)

The term `support` is also used in the method linked above. For the `avg`s, this should be the total number of train and test examples. The reason you're seeing numbers differ for different number of folds (`f`) is because we drop intents that have less than `f` examples. This is to allow for the test data to be different for each fold. So if you have at least `3` examples for each intent and `f=3` you should see `support` for these `avg`s be equal to your number of train and test examples.

Does that clear things up?

**Comment by ericZYZ** [comment]
Hi thank you for the detailed explanation. I will take a look at the methods from sklearn. However I'm not able to find a `report.json` file. I'm using Rasa 2.0.2

**Comment by koernerfelicia** [comment]
Oh, my bad, I'm sorry! I didn't type out the full name, I meant `intent_report.json` (will edit). Does this mean we can close the issue? Or do you have any other concerns regarding this?

**Comment by ericZYZ** [comment]
Hi I just checked `intent_report.json` and there is no `'accuracy'` metric. If you are referring to `precision`, they don't match the results printed in the terminal either. Am I missing something here?

**Comment by koernerfelicia** [comment]
`accuracy` is only included if `micro-avg` isn't included, you should have one or the other (see the [docs](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.classification_report.html?highlight=classification_report) for `classification_report`).

**Comment by TyDunn** [comment]
Please reopen if this is still an issue

**Comment by dhruvbhimani-2512** [comment]
Hey @koernerfelicia, I am having a different issue with number of supports.

I am comparing two NLU pipelines using `rasa test` command. When I evaluate final report for entity classification, total supports present in my testing data `test.md` and in the entity report do not match. I am always getting ~1 to 3 additional supports than test data in the report.
